### ISSUE TITLE
3.4.1 compile errors on MacOS

### ISSUE THREAD

**Issue by genehack** [sass/perl-libsass#28]
This is on:

```
% uname -a 
Darwin irving.local 16.4.0 Darwin Kernel Version 16.4.0: Thu Dec 22 22:53:21 PST 2016; root:xnu-3789.41.3~3/RELEASE_X86_64 x86_64 i386 MacBookPro11,1 Darwin
```

which is "10.12.3" in the "About this Mac" dialog. 

I run `perl Makefile.PL && make` and it fails during the compile.

I reran `make > out.txt 2>&1`, and will attach the resulting file here: [out.txt](https://github.com/sass/perl-libsass/files/775179/out.txt)

**Comment by mgreter** [comment]
Thanks @genehack for the bug report. Good news is that I can reproduce the issue on my Yosemite virtualbox machine (you have no idea how long it took me to get this working). Next good news is that you should be able to get a working version with `perl Makefile.PL --no-plugins`, as noted in the release notes at https://github.com/sass/perl-libsass/releases/tag/3.4.1.

None the less I would like to get this working by default also on Mac OSX. My problem here is that the perl build toolchain is not flexible enough to fully support this use-case. So I had to hack and hook into EUMM to get it working. It might have been a bit premature to release and enable this already and I appologize for the inconvenience, but I also hope you understand that this is the only way for me to really move this forward. I think once it also works on mac, you will be happy with the newly available plugins!

Since I can reproduce it here I'm positive that I can come up with a fix!

**Comment by mgreter** [comment]
http://www.cpantesters.org/cpan/report/07d9547c-d95a-11e6-91f1-a230ca8448b3

**Comment by mgreter** [comment]
The actual problem is that the build file invokes cc instead of g++ when linking the shared libsass library, therefore lacking the c++ symbols. The perl dll was always statically linked against libsass, but this is no longer "desired" with the additional plugins, because each plugins also needs to link against libsass. For now I kept the main perl module dll statically linked and only create a shared libsass dll for the plugins. That is why the default build currently fails on OSX when plugins are enables (which is the default ATM). Worst case scenario would be to just disable the plugins again if there is a wall I can't navigate around!

**Comment by mgreter** [comment]
@genehack I've added a fix in https://github.com/sass/perl-libsass/commit/05d614442e97de2ba587fc5bba04e3a326781825. I hope this does not regress on any other platform. Can you try to build it from current development branch and verify that it works by calling `psass --list-plugins` after `make install`? You should get `--glob-plugin` and `--math-plugin`.

If you can confirm I will package a 3.4.2 release with these fixes!

**Comment by mgreter** [comment]
Since all my test were positive I went on and released 3.4.2!
Hopefully http://matrix.cpantesters.org/?dist=CSS-Sass agrees with the fixes!

**Comment by genehack** [comment]
@mgreter Sorry, finally got a chance to go back and check this out - 3.4.2 installed cleanly here. Thanks!
